shell/ev-application.c

```diff
@@ -419,9 +419,18 @@ on_register_uri_cb (GObject      *source_object,
 	g_variant_builder_add (&builder, "{sv}",
 	                       "display",
 	                       g_variant_new_string (gdk_display_get_name (gdk_screen_get_display (data->screen))));
-	g_variant_builder_add (&builder, "{sv}",
-	                       "screen",
-	                       g_variant_new_int32 (gdk_x11_screen_get_screen_number (data->screen)));
+
+	if (GDK_IS_X11_DISPLAY (gdk_display_get_default())){
+		g_variant_builder_add (&builder, "{sv}",
+		                     "screen",
+		                      g_variant_new_int32 (gdk_x11_screen_get_screen_number (data->screen)));
+	}
+	else{
+		/*Do not crash on wayland, use the first monitor for now*/
```

Comment:
It's not the monitor but the X screen, which is deprecated in GDK and a X11-only concept so using `1` here -- or even nothing -- is fine I guess.

Reply:
Wondering if this should actually be zero: there is only one screen even in x11 since at least 3.22 and it is numbered 0

Reply:
Second force-push changes this to 0, tested with no problems on my end

Reply:
Marking resolved as suggestion has been implemented and the screen number has been set to zero